### PR TITLE
[keyring-controller] Remove reliance upon the `PreferencesController`

### LINKED ISSUE #3699: [keyring-controller] Remove reliance upon the `PreferencesController`
The `KeyringController` currently depends upon four methods from the `PreferencesController` being passed in as constructor arguments. These methods are called to synchronize state, and to update preferences in response to certain keyring actions.

Essentially those operations don't belong in the `KeyringController`. It's not responsible for the `PreferencesController` state, the `PreferencesController` is. Conventionally a controller manages its own state, and that's it. The conventional approach here would be to update the `PreferencesController` to listen for relevant keyring related events, and react to them as appropriate internally. This keeps responsibilities clearly delineated between each controller. Event-based syncing is also safer than active syncing, in that we don't have to be concerned about forgetting to sync when we need to.

### PR DESCRIPTION
**Pull request by MajorLift**
## Explanation

Removes all `PreferencesController` callbacks from `KeyringController` class, methods, and tests.

## References

- Closes #3699 

### Removed

- **BREAKING:** Remove `updateIdentities`, `syncIdentities`, `setSelectedAddress`, `setAccountLabel` from constructor options of `KeyringController` class.

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: packages/keyring-controller/src/KeyringController.ts [hunk 1]
```diff
@@ -168,6 +168,26 @@ export type KeyringControllerQRKeyringStateChangeEvent = {
...
+export type KeyringControllerSetSelectedAddressEvent = {
+  type: `${typeof name}:setSelectedAddress`;
+  payload: [selectedAddress: string];
+};
+
+export type KeyringControllerSetAccountLabelEvent = {
+  type: `${typeof name}:setAccountLabel`;
+  payload: [address: string, label: string];
+};
+
```
Comment (mikesposito) [inline packages/keyring-controller/src/KeyringController.ts:190]: `PreferencesController` has now responsibilities over syncing identities with the KeyringController:
https://github.com/MetaMask/core/blob/79ebb87ec9559b831e74de40efbb2d8624d4b774/packages/preferences-controller/src/PreferencesController.ts#L235

So we can simply remove all calls to `updateIdentities`, `syncIdentities`, `setSelectedAddress`, `setAccountLabel` and the corresponding constructor options from KeyringController, without adding events to the `messenger`

Comment (mikesposito) [reply packages/keyring-controller/src/KeyringController.ts:190]: Uhm, I'm not sure about `setAccountLabel` though. I don't think that's handled by PreferencesController at the moment, but I also don't feel like it should be KeyringController's responsibility to change the account label or select a different address

Comment (mikesposito) [reply packages/keyring-controller/src/KeyringController.ts:190]: Right, `setSelectedAddress` and `setAccountLabel` are now part of `AccountsController`, that's why they are not managed by `PreferencesController` (even though this is still the case on the extension). 

Given that the intention is to give this responsibility to `AccountsController`, we should keep  `setSelectedAddress` and `setAccountLabels` in place. 

We'll probably have to add some events to allow `AccountsController` to know that a label should be changed or an address should be selected - but I'm not even sure of that, as from what I can see these two methods are only used in specific cases for QRKeyring, which might go away with https://github.com/MetaMask/core/issues/3786

Comment (MajorLift) [reply packages/keyring-controller/src/KeyringController.ts]: Applied in https://github.com/MetaMask/core/pull/3853/commits/801026c0b66d557de43372243ba5181c0d175036, https://github.com/MetaMask/core/pull/3853/commits/806fd9b696abb1aa088bb9accce442b2852187ac!

File: packages/keyring-controller/CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Removed
+
+- **BREAKING:** Remove `updateIdentities`, `syncIdentities`, `setSelectedAddress`, `setAccountLabel` from constructor options of `KeyringController` class.
```
Comment (mcmire) [inline packages/keyring-controller/CHANGELOG.md:12]: Do we also want to mention the methods that were changed along with the fact that that they no longer call these functions?

**Review by mcmire — COMMENTED** [review]
Looks good! Wondering if we need to add a bit more information to the changelog.

File: packages/keyring-controller/CHANGELOG.md (hunk 2 above)
Comment (MajorLift) [reply packages/keyring-controller/CHANGELOG.md:12]:
```suggestion
- **BREAKING:** Remove `updateIdentities`, `syncIdentities`, `setSelectedAddress`, `setAccountLabel` callbacks, which previously updated `PreferencesController` state, from constructor options of `KeyringController` class. ([#3853](https://github.com/MetaMask/core/pull/3853))
  - Class methods `addNewAccount`, `addNewAccountForKeyring`, `createNewVaultAndRestore`, `createNewVaultAndKeychain`, `importAccountWithStrategy`, `restoreQRKeyring`, `unlockQRHardwareWalletAccount`, `forgetQRDevice` no longer directly updates `PreferencesController` state by calling the `updateIdentities` callback.
  - Class method `submitPassword` no longer directly updates `PreferencesController` state by calling the `syncIdentities` callback.
  - Class method `unlockQRHardwareWalletAccount` no longer directly updates `PreferencesController` state by calling the `setAccountLabel`, `setSelectedAddress, callbacks.
```

Comment (MajorLift) [reply packages/keyring-controller/CHANGELOG.md:12]: @mcmire How does this look?

Comment (mcmire) [reply packages/keyring-controller/CHANGELOG.md:12]: @MajorLift Yeah, that looks good!

**Review by mikesposito — APPROVED** [review]
Looks good!

**Pull request merged**